### PR TITLE
Repare tests

### PR DESCRIPTION
**Pull request by mdellweg**
* Rerecord domain
* Ignore smart_proxy name in domain test playback
* Also move domain_dns_proxy variable to server_vars.yml as it is volatile.

### CODE REVIEW

File: test/test_playbooks/domain_vars.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
-domain_name: "example.org"
-domain_dns_proxy: "centos7-devel.example.com"
+domain_name: "test-domain.org"
```
Comment (evgeni) [inline test/test_playbooks/domain_vars.yml]: I'd prefer not to use a non-example domain here.

File: test/vcr_python_wrapper.py [hunk 2]
```diff
@@ -35,6 +35,12 @@ def body_json_l2_matcher(r1, r2):
         return r1.body == r2.body
 
 
+def domain_query_matcher(r1, r2):
+    if r1.path == '/api/smart_proxies' and r2.path == '/api/smart_proxies':
+        return [q for q in r1.query if q[0] != 'search'] == [q for q in r2.query if q[0] != 'search']
```
Comment (evgeni) [inline test/vcr_python_wrapper.py]: Should we use sorted() here to ignore the param order?

Comment (mdellweg) [reply test/vcr_python_wrapper.py]: according to vcrpy docs it is ordered

> query: The parsed query string of the request. Sorted list of name, value pairs.

https://vcrpy.readthedocs.io/en/latest/advanced.html?highlight=query#advanced-features

File: test/test_playbooks/domain_vars.yml (hunk 1 above)
Comment (mdellweg) [reply test/test_playbooks/domain_vars.yml]: OK. I used that to make sure, it did not fail the matching.

File: test/vcr_python_wrapper.py (hunk 2 above)
Comment (evgeni) [reply test/vcr_python_wrapper.py]: Cool. If vcr does that for us, we're good

File: test/test_playbooks/domain_vars.yml (hunk 1 above)
Comment (evgeni) [reply test/test_playbooks/domain_vars.yml]: Right. We have a few possibilities here (see https://tools.ietf.org/html/rfc6761), but we shouldn't use a domain that could be owned by somebody else.

Comment (mdellweg) [reply test/test_playbooks/domain_vars.yml]: Did i? Anyway, changing it back.

File: test/test_playbooks/domain_vars.yml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 ---
-domain_name: "example.org"
-domain_dns_proxy: "centos7-devel.example.com"
+domain_name: "example.com"
```
Comment (evgeni) [inline test/test_playbooks/domain_vars.yml]: I think we used the dot org in the past to differ more explicitly from a default forklift setup (which uses `$(hostname -s).example.com` as the primary domain name). But I'm fine with either.

**Pull request merged**